independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -188,6 +188,13 @@ public class BeanDeployment {
                     if (nonbindingMembers == null) {
                         nonbindingMembers = Collections.emptySet();
                     }
+                    for (MethodInfo mi : classInfo.methods()) {
+                        if (mi.returnType().kind().equals(Type.Kind.ARRAY) && !nonbindingMembers.contains(mi.name())) {
+                            throw new DefinitionException("Qualifier annotation '" + classInfo + "' contains a method '"
```

Comment:
I wonder if we should detect annotation-valued members as well...

Also I think that for annotations it's more common to use the term `member` instead of `method`.

Reply:
> I wonder if we should detect annotation-valued members as well...

Right, I forgot spec also mentions those.
I'll add that as well.

> Also I think that for annotations it's more common to use the term member instead of method.

Fair point, will change the message :+1:

Reply:
I have added a check for the annotation-valued members as well as automated tests for both - discovered and registered qualifiers.
The logic is bit more complex now so I extracted it into a separate method.